run/invoker.go

```diff
@@ -0,0 +1,220 @@
+package run
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	_ "net/http/pprof"
+	"os"
+	"runtime/debug"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"github.com/testground/sdk-go/network"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+)
+
+const (
+	// These ports are the HTTP ports we'll attempt to bind to. If this instance
+	// is running in a Docker container, binding to 6060 is safe. If it's a
+	// local:exec run, these ports belong to the host, so starting more than one
+	// instance will lead to a collision. Therefore we fallback to 0.
+	HTTPPort         = 6060
+	HTTPPortFallback = 0
+
+	StateInitializedGlobal   = sync.State("initialized_global")
+	StateInitializedGroupFmt = "initialized_group_%s"
+)
+
+// HTTPListenAddr will be set to the listener address _before_ the test case is
+// invoked. If we were unable to start the listener, this value will be "".
+var HTTPListenAddr string
+
+// InitContext encapsulates a sync client, a net client, and global and
+// group-scoped seq numbers assigned to this test instance by the sync service.
+//
+// The states we signal to acquire the global and group-scoped seq numbers are:
+//  - initialized_global
+//  - initialized_group_<id>
+type InitContext struct {
+	SyncClient sync.Interface
+	NetClient  *network.Client
+	GlobalSeq  int64
+	GroupSeq   int64
+}
+
+// init can be safely invoked on a nil reference.
+func (ic *InitContext) init(runenv *runtime.RunEnv) {
+	var (
+		grpstate  = sync.State(fmt.Sprintf(StateInitializedGroupFmt, runenv.TestGroupID))
+		client    = sync.MustBoundClient(context.Background(), runenv)
+		netclient = network.NewClient(ic.SyncClient, runenv)
```

Comment:
Not sure how this is going to work on a `nil` `ic` - seems like we define `ic.SyncClient` below - maybe change to `client`, or?

Reply:
Oh yeah, good catch; I shuffled some code around, and this happened.